e2e/e2e-script.sh

```diff
@@ -0,0 +1,181 @@
+#!/bin/bash
+
+echo "Starting e2e tests"
+
+# TODO 1: Probably move the helper functions to some other file to imrpove readability
+
+addJsonToFile() {
+    k=$1; v=$2
+    jq -r --arg key $k --arg value $v '. + { ($key) : $value}' < fields.json > dummy.json && mv dummy.json fields.json
+}
+
+getAgentPoolProfileValues() {
+    declare -a properties=("mode" "name" "nodeImageVersion")
+
+    for property in "${properties[@]}"; do
+        value=$(az aks show -n $CLUSTER_NAME -g $RESOURCE_GROUP_NAME | jq -r .agentPoolProfiles[].${property})
+        addJsonToFile $property $value
+    done
+}
+
+getFQDN() {
+    fqdn=$(az aks show -n $CLUSTER_NAME -g $RESOURCE_GROUP_NAME | jq -r '.fqdn')
```

Comment:
calling az aks show multiple times may have performance issue. We can save the output in a file and keep using the same file for getting properties.

Reply:
that makes a lot of sense since I am calling az aks show ~10 times. should have thought of that, ill update the logic.